jira-config/jira-config-platform/src/main/java/org/xwiki/contrib/jira/config/internal/JIRAConfigurationMigrator.java

```diff
@@ -0,0 +1,139 @@
+     */
+    public void migrate()
+    {
+        XWikiContext context = contextProvider.get();
+        XWiki xwiki = context.getWiki();
+        DocumentReference migrationDocumentRef =
+            new DocumentReference(MIGRATION_STATUS_DOCUMENT_REFERENCE, new WikiReference(context.getWikiId()));
+
+        try {
+            if (!xwiki.exists(migrationDocumentRef, context)) {
+                logger.info("Running migration of JIRA configuration");
+
+                XWikiDocument basicAuthDoc =
+                    xwiki.getDocument(BasicAuthJIRAAuthenticatorFactory.BASIC_AUTH_CONFIG_REFERENCE, context).clone();
+                XWikiDocument jiraServerDoc =
+                    xwiki.getDocument(new LocalDocumentReference(JIRA, "JIRAConfig"), context).clone();
+                List<BaseObject> jiraServerObjs =
+                    jiraServerDoc.getXObjects(new LocalDocumentReference(JIRA, "JIRAConfigClass"));
+                for (BaseObject obj : jiraServerObjs) {
+                    if (obj == null) {
+                        continue;
+                    }
+                    String serverId = obj.getStringValue(PROPERTY_ID);
+                    String username = obj.getStringValue(PROPERTY_USERNAME);
+                    String password = obj.getStringValue(PROPERTY_PASSWORD);
+
+                    BaseObject jiraServerObj = jiraServerObjs.stream()
+                        .filter(o -> o != null && StringUtils.equals(o.getStringValue(PROPERTY_ID), serverId))
+                        .findFirst().orElseThrow();
+                    jiraServerObj.removeField(PROPERTY_USERNAME);
+                    jiraServerObj.removeField(PROPERTY_PASSWORD);
+
+                    if (StringUtils.isEmpty(username)) {
+                        // server without authentication
+                        jiraServerObj.setStringValue(PROPERTY_AUTHENTICATION_TYPE, "noAuth");
+                    } else {
+                        jiraServerObj.setStringValue(PROPERTY_AUTHENTICATION_TYPE,
+                            BasicAuthJIRAAuthenticatorFactory.HINT);
+                        BaseObject basicAuthObj = basicAuthDoc
+                            .newXObject(BasicAuthJIRAAuthenticatorFactory.BASIC_AUTH_DATA_CLASS_REFERENCE, context);
+                        basicAuthObj.setStringValue(PROPERTY_ID, serverId);
+                        basicAuthObj.setStringValue(PROPERTY_USERNAME, username);
+                        basicAuthObj.setStringValue(PROPERTY_PASSWORD, password);
+                    }
+                }
+                // TODO set correct user for save...
+                xwiki.saveDocument(basicAuthDoc, DOC_SAVE_MESSAGE, context);
+                xwiki.saveDocument(jiraServerDoc, DOC_SAVE_MESSAGE, context);
+                // Juste create a single document to save the fact that we run the migration
+                xwiki.saveDocument(xwiki.getDocument(migrationDocumentRef, context).clone(), context);
```

Comment:
Currently the document is marked as saved by XWikiGuest, what would be the recommended way for this ?